codegen/template.go

```diff
@@ -637,11 +662,26 @@ func (t *Template) GenerateMainFile(
 	}, nil
 }
 
-func (t *Template) execTemplateAndFmt(templName string, filePath string, data interface{}) error {
+func (t *Template) execTemplateAndFmt(
+	templName string,
+	filePath string,
+	data interface{},
+	copyrightHeader string,
```

Comment:
Strictly speaking, it is `copyRightHeaderFile` not the "copyRightHeader" string itself.

Also, we should allow the file to be `""` and set "// Generated by Zanzibar ..." by default.

To be more robust, zanzibar should throw error if the header file doesn't have `//` at the beginning of each line.